artifacts/notification-as-cloud-event.yaml

```diff
@@ -0,0 +1,287 @@
+openapi: 3.0.3
+info:
+  title: Event Notification using CloudEvents specifications
+  description: |
+    The event notification endpoint is used by the API server to notify the API consumer that an event occurred.  The notification is the message posted on listener side.
+
+    # Introduction
+
+    A lot of CAMARA APIs offer the capability to API consumer to receive events.
+    Event data are defined in each API definition but in order to provide consistency across CAMARA APIs and to increase 
+    interoperability we will use [cloudevents](https://cloudevents.io/) specifications. In particular, every CAMARA Event will 
+    be defined using [cloudevents-json-format](https://github.com/cloudevents/spec/blob/main/cloudevents/formats/json-format.md)
+   
+    # Relevant terms and definitions
+
+      * **Occurrence** : An "occurrence" is the capture of a statement of fact during the operation of a software system.
+  
+      * **Event**: An "event" is a data record expressing an occurrence and its context. Events are routed from an 
+    event producer (the source) to interested event consumers.
+    
+      * **Producer**: The "producer" is a specific instance, process or device that creates the data structure 
+    describing the CloudEvent.
+  
+      * **Source**:  The "source" is the context in which the occurrence happened. In a distributed system it might 
+    consist of multiple producers. If a source is not aware of CloudEvents, an external producer creates 
+    the CloudEvent on behalf of the source.
+  
+      * **Consumer**: A "consumer" receives the event and acts upon it. It uses the context and data to execute some 
+    logic, which might lead to the occurrence of new events.
+  
+      * **Data**:  Domain-specific information about the occurrence (i.e. the payload). This might 
+    include information about the occurrence, details about the data that was changed, or more.
+  
+    
+    # API Functionality   
+      
+    Only one endpoint/operation is provided: `POST /your_webhook_notification_url`
+    
+    This endpoint describes the event notification received on subscription listener side when the event occurred. A detailed description of the event notification is provided in the CAMARA API design guideline document.
+  
+  
+
+  termsOfService: http://swagger.io/terms/
+  contact:
+    email: project-email@sample.com
+  license:
+    name: Apache 2.0
+    url: https://www.apache.org/licenses/LICENSE-2.0.html
+  version: 0.1.0-wip
+externalDocs:
+  description: Product documentation at CAMARA
+  url: https://github.com/camaraproject/
+security:
+  - oAuth2ClientCredentials: []
```

Comment:
Aligment would be required after resolution of [PR#57](https://github.com/camaraproject/Commonalities/pull/57/files)

Reply:
The API design guidelines allow the API consumer to have no security scheme (authorisation scheme) for event notifications. So no security also needs to be an option here.

Reply:
Hi @eric-murray ,
I added : "Security is the responsibility of the API provider. The API provider must include its requirements in the security section of this specification or remove this section if it does not support any authentication scheme.
"
What do you think about?

Reply:
@patrice-conil
I've seen no authentication scheme specified explicitly in a Swagger as `{}`
So I was expecting to see something like the following within the path definition:
```
      security:
        - oAuth2ClientCredentials: []
        - {}
```
In other words, either client credentials or no authorisation is allowed, but nothing else. When we need to send a notification, we would know which to use depending upon whether the client originally gave us a `notificationAuthToken` or not.

Anyway, this is just what I was expecting to see. If the client is expecting to receive notifications using some other authorisation scheme, they will be disappointed when they do not receive any.

Reply:
I missed this one! Done!

Reply:
Hi, 

@patrice-conil, @eric-murray regarding this thread. 

Two comments:
- scope: think we need to consider a scope for the use of this interface (e.g. notifications:create)

- security: think we should not specify security scheme within the interface and describe something like -> 
This interface can be used with securization (Authorization Header) or without it depending on the indication of `webhook.notificationAuthToken` in the involved procedure that generates associated event(s), applicable for both Instance-based or Resource-Based subscription models. So as when `webhook.notificationAuthToken` is not provided, no securization header will be indicated by menas of this interface.


Rationale: the way the API consumer generates the "notificationAuthToken" is on its discretion.

Reply:
There is an open question that, if the API consumer provides an OAuth server for their notification endpoint and advertises that in their version of this YAML, should the API implementor use this to get a token, rather than mandate that the API consumer generate a token and provide this when they create the subscription?

If that is possible, then they will need to define the required scope (as well as providing the API implementor with a client id / secret). But I think the scope name is up to them. I don't see that CAMARA should impose their scope naming convention on the API consumer's own OAuth service.

Reply:
@patrice-conil, @Eric regarding this point: Indicating `notificationsBearerAuth` scheme really shows how the notificationAuthToken is considered by Telco Operator in case of its reception in API request or explicit suscription.

With `oAuth2ClientCredentials: []`, is meant that the notificationAuthToken is an oAuth 2.0 client credentials token, and that is misleading because how that notificationAuthToken is generated by API consumer is not controlled neither decided by Telco Operator, it is up to the API consumer its generation.

Reply:
@PedroDiez, @eric-murray 
Suggestion from Pedro is integrated in last commit, Eric do you agree with that ?